### PR TITLE
README.md: Improve documentation

### PR DESCRIPTION
**Pull request by agherzan**
Signed-off-by: Andrei Gherzan <andrei@resin.io>

@petrosagg @pcarranzav @telphan @floion

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,51 +1,62 @@
...
+
+## Docker containers versus git repository
+The releases for resinhup are marked in this git repository as git tags. As well the git repository includes Dockerfiles for each resinhup supported resin board. For example there is a Dockerfile called _Dockerfile.raspberrypi3_ which is the Dockerfile for creating a resinhup docker image for [Raspberry Pi 3](https://www.raspberrypi.org/blog/raspberry-pi-3-on-sale/) boards. We currently upload our resinhup docker images to resin registry (registry.resinstaging.io). We do this because old resin devices were using docker 1.4 which can't pull from registry v2 dockerhub.
+
+For each resinhup release (tag) there will be a set of docker images with the same tag uploaded to the above mentioned registry. The images have the name format: _resinhup-<board-slug>_. The full docker images URL format becomes _registry.resinstaging.io/resinhup/resinhup-<board slug>_.
```
Comment (pcarranzav) [inline README.md]: Can you format the image names as `code`? Otherwise the board-slug doesn't show up

File: README.md [hunk 2]
```diff
@@ -1,51 +1,62 @@
...
+
+Example. For release 1.0 (git tag 1.0), resinhup supports the following device slugs: beaglebone-black, intel-nuc, raspberry-pi, raspberry-pi2 and raspberrypi3. For each slug there is a corresponding Dockerfile: Dockerfile.beaglebone-black, Dockerfile.intel-nuc, Dockerfile.raspberry-pi, Dockerfile.raspberry-pi2 and Dockerfile.raspberrypi3. For each Dockerfile there are docker image pushed to the registry:
+registry.resinstaging.io/resinhup/resinhup-raspberrypi3       1.0  c324b00459f3        2 days ago          241.4 MB
```
Comment (pcarranzav) [inline README.md]: Same with this list

File: README.md [hunk 3]
```diff
@@ -1,51 +1,63 @@
...
+
+## Resinhup architecture
+Currently there are 3 components involved in updating a device:
++ resinhup (docker images)
++ run-resinhup.sh (bash wrapper)
++ run-resinhup-ssh.sh (bash wrapper)
+
+### resinhup
+This component is distributed from this repository as docker images (explained above). The overall workflow of the tool is:
+![Minion](https://github.com/resin-os/resinhup/blob/master/images/resinhup-workflow.png)
```
Comment (pcarranzav) [inline README.md]: I think you can use a relative path for the image, that way it will also show up in branches

File: README.md [hunk 4]
```diff
@@ -1,51 +1,63 @@
...
++ takes care of all the prerequisites 
++ adds support for supervisor update
++ pull resinhup image
++ run resinhup container
++ if updater is successful, reboots the board
+
+### run-resinhup-ssh.sh 
+This is a tool which:
+
++ Uploads over ssh all the needed tools from meta-resin (updare-resin-supervisor, run-resinhup.sh and resin-device-progress).
```
Comment (pcarranzav) [inline README.md]: Typo `updare-resin-supervisor -> update-resin-supervisor`

Comment (agherzan) [inline README.md]: Fixing.

File: README.md (hunk 3 above)
Comment (agherzan) [inline README.md]: Fixing.

File: README.md (hunk 2 above)
Comment (agherzan) [inline README.md]: Fixed.

File: README.md (hunk 1 above)
Comment (agherzan) [inline README.md]: Fixed.

File: README.md [hunk 5]
```diff
@@ -1,51 +1,66 @@
 # Resinhup
 Tool for resin host OS updates.
```
Comment (pcarranzav) [inline README.md]: Might be good to add a short description of the workflow as an intro. i.e. something like "This tool is a docker image that downloads an update bundle that replaces the HostOS for a Resin device, updating both boot partition and rootfs (using an inactive partition)".

Comment (agherzan) [inline README.md]: Makes sense.

**Comment by pcarranzav** [comment]
LGTM

File: README.md [hunk 6]
```diff
@@ -1,51 +1,66 @@
...
 
-### Run container automatically
-Run the container automatically:
-```sh
-[host] $ rce run -ti --privileged --rm --net=host --volume /:/host resinhup
-[host] $ reboot
-```
+The images taged as _latest_ are following the HEAD of master branch.
+
+Hint: there is script helper called [docker-build-and-push.sh](https://github.com/resin-os/resinhup/blob/master/scripts/docker-build-and-push.sh) for pushing images to registry. Check its help.
```
Comment (floion) [inline README.md]: typo. should be "there is a script"

File: README.md [hunk 7]
```diff
@@ -1,51 +1,66 @@
...
+
+## How to use
+Use run-resinhup.sh [wrapper](https://github.com/resin-os/meta-resin/blob/master/meta-resin-common/recipes-support/resinhup/resinhup/run-resinhup.sh) developed in [meta-resin](https://github.com/resin-os/meta-resin). Check _run-resinhup.sh_ help message for all the configuration you can use. Make sure the _run-resinhup.sh_ along with _update-resin-supervisor_ and _resin-device-progress_ scripts are updated.
+
+Pro Hint: In order to make sure these scripts are updated and run resinhup multiple devices **admins** can use a wrapper on top of _run-resinhup.sh_ called [run-resinhup-ssh.sh](https://github.com/resin-os/meta-resin/blob/master/scripts/resinhup/run-resinhup-ssh.sh). This wrapper is not intended for public use as it requires SSH access to the devices over VPN along with admin permissions for API queries. Check _run-resinhup-ssh.sh_ help message for all the configuration you can use.
```
Comment (floion) [inline README.md]: should this be: "these scripts are updated and ran" ?

Comment (floion) [inline README.md]: There should also be a comma somewhere in there, the semantics are a bit strange.

Comment (floion) [inline README.md]: should this be: "these scripts are updated and ran" ?

Comment (pcarranzav) [inline README.md]: Nope, "run" is the past participle

File: README.md (hunk 6 above)
Comment (agherzan) [inline README.md]: Fixed.

File: README.md (hunk 7 above)
Comment (agherzan) [inline README.md]: Indeed. "are run" is using the past participle form of the verb "to run". "Ran" as past tense is used when not in combination with another verb.

Comment (agherzan) [inline README.md]: Even so I do admit the phrase is not that clear. I will redo.

File: README.md [hunk 8]
```diff
@@ -1,51 +1,66 @@
...
+
+Hint: there is a script helper called [docker-build-and-push.sh](https://github.com/resin-os/resinhup/blob/master/scripts/docker-build-and-push.sh) for pushing images to registry. Check its help.
+
+## How to use
+Use run-resinhup.sh [wrapper](https://github.com/resin-os/meta-resin/blob/master/meta-resin-common/recipes-support/resinhup/resinhup/run-resinhup.sh) developed in [meta-resin](https://github.com/resin-os/meta-resin). Check _run-resinhup.sh_ help message for all the configuration you can use. Make sure the _run-resinhup.sh_ along with _update-resin-supervisor_ and _resin-device-progress_ scripts are updated.
+
+Pro Hint: In order to make sure these scripts are updated and be able to run resinhup on multiple devices (batch/fleet updates), **admins** can use a wrapper on top of _run-resinhup.sh_ called [run-resinhup-ssh.sh](https://github.com/resin-os/meta-resin/blob/master/scripts/resinhup/run-resinhup-ssh.sh). This wrapper is not intended for public use as it requires SSH access to the devices over VPN along with admin permissions for API queries. Check _run-resinhup-ssh.sh_ help message for all the configurations you can use.
```
Comment (Page-) [inline README.md]: `are updated and be able` -> `are updated and able`

Comment (agherzan) [inline README.md]: Fixed.

**Comment by petrosagg** [comment]
lgtm

**Pull request merged**